webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
-            publish = true;
-          }
+      arrayToClassify.add(id);
+      if (publish) {
+        scc.publish(arrayToClassify); // set etat du site a 1
+      } else {
+        scc.dePublish(arrayToClassify);
+      }
+    }
 
-          scc.updateClassification(id, arrayTopic);
+    Collection<SiteDetail> listeSites = scc.getAllWebSite();
+    request.setAttribute("ListSites", listeSites);
+    request.setAttribute("BookmarkMode", Boolean.valueOf(scc.isBookmarkMode()));
 
-          arrayToClassify.add(id);
-          if (publish) {
-            scc.publish(arrayToClassify); // set etat du site a 1
-          } else {
-            scc.dePublish(arrayToClassify);
-          }
-        }
+    return "/webSites/jsp/manage.jsp";
+  }
 
-        Collection<SiteDetail> listeSites = scc.getAllWebSite();
-        request.setAttribute("ListSites", listeSites);
-        request.setAttribute("BookmarkMode", Boolean.valueOf(scc.isBookmarkMode()));
+  private String classification(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
 
-        destination = "/webSites/jsp/manage.jsp";
-      } else if (function.startsWith("design.jsp")) {
-        // Action = newSite the firt time, never null
-        String action = request.getParameter("Action");
-        String id = request.getParameter("Id"); // jamais null sauf en creation ou en
-        // Retrieve currentPath parameter : null when creating a webSite
-
-        String currentPath = request.getParameter("path");
-        if (currentPath != null) {
-          currentPath = doubleAntiSlash(currentPath);
-        }
+    String action = request.getParameter("Action");
+    String id = request.getParameter("TopicId");
+    String linkedPathString = request.getParameter("Path");
 
-        // ADD NEW SITE -------------------------------------------------------------
-        if ("newSite".equals(action)) {
-          // Filled at first access then null
-          String nomSite = request.getParameter("nomSite");
-          // Filled at first access then null
-          String description = request.getParameter("description");
-          // Filled at first access then null
-          String nomPage = request.getParameter("nomPage");
-          String tempPopup = request.getParameter("popup");
-          // Filled at first creation then null
-          String listeIcones = request.getParameter("ListeIcones");
-          // = en cas de new Site ou de classifySite
-          String listeTopics = request.getParameter("ListeTopics");
-          // Retrieve positions
-          String positions = request.getParameter("Positions");
-
-          int popup = 0;
-          if ((tempPopup != null) && (tempPopup.length() > 0)) {
-            popup = 1;
-          }
+    Collection<SiteDetail> listeSites = scc.getAllWebSite();
+    request.setAttribute("ListSites", listeSites);
+    request.setAttribute("CurrentFolder", scc.getSessionTopic());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "CurrentFolder" 11 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)